app/src/main/java/com/example/bottomnav/Parse.java

```diff
@@ -50,7 +60,39 @@ public Parse(char[] code) throws Exception {
                 }
             }
         }
-        associateIDStruct(comments);
+        parseComments(comments);
+    }
+
+    @RequiresApi(api = Build.VERSION_CODES.O)
+    public void decode(int canID, byte[] payload) {
+        String payloadDataType = getConst(canID).payLoadDataType;
+        switch (payloadDataType){
+            case "single":
+                payloadMap.put(canID,
+                        ByteBuffer.wrap(payload).order(ByteOrder.LITTLE_ENDIAN).getFloat());
+                break;
+            case "struct":
+                ArrayList<StructContents> structContents = getStruct(canID);
+                int incr = payload.length / structContents.size();
```

Comment:
You shouldn't treat structs as repeating - once you parse an entire struct you should discard the rest of the data (if any).

If you wanted to be really robust and thorough, you can generate an error type (or error annotation) if there is leftover data. Similarly, think about how to handle when there isn't enough data.